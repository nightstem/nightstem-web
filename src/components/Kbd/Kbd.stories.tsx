import type { Meta, StoryObj } from '@storybook/nextjs-vite';

import KbdComponet, { type KbdProps } from '@/components/Kbd';

const meta = {
  component: KbdComponet,
  tags: ['autodocs'],
} satisfies Meta<KbdProps>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Kbd: Story = {
  args: {
    children: 'Ctrl',
  },
};
