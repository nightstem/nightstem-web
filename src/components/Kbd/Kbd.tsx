import cx from 'classnames';
import { DetailedHTMLProps, HTMLAttributes } from 'react';

export type KbdProps = DetailedHTMLProps<
  HTMLAttributes<HTMLElement>,
  HTMLElement
>;

export const Kbd = ({ className, ...props }: KbdProps) => (
  <kbd
    {...props}
    className={cx(
      'rounded shadow-2xl border border-foreground/20 px-1',
      className,
    )}
  />
);

export default Kbd;
