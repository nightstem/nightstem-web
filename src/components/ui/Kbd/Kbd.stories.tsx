import type { Meta, StoryObj } from '@storybook/nextjs-vite';

import KbdComponent, { type KbdProps } from '@/components/ui/Kbd';

const meta = {
  component: KbdComponent,
  tags: ['autodocs'],
} satisfies Meta<KbdProps>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Kbd: Story = {
  args: {
    children: 'Ctrl',
  },
};
