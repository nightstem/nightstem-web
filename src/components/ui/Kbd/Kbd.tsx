import cx from 'classnames';
import { DetailedHTMLProps, HTMLAttributes } from 'react';

export type KbdProps = DetailedHTMLProps<
  HTMLAttributes<HTMLElement>,
  HTMLElement
>;

const Kbd = ({ className, ...props }: KbdProps) => (
  <kbd
    {...props}
    className={cx(
      'rounded border border-foreground/20 p-1.5 leading-[50%] shadow-2xl',
      className,
    )}
  />
);

export default Kbd;
