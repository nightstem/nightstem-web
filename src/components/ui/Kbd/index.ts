export * from './Kbd';
export { default } from './Kbd';
