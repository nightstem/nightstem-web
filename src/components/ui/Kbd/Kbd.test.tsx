import { axe } from 'vitest-axe';
import { render } from '@testing-library/react';

import Kbd, { type KbdProps } from '@/components/ui/Kbd';

const defaultProps: KbdProps = {
  children: 'Ctrl',
};

it('does not have any accessibility violations', async () => {
  const { container } = render(<Kbd {...defaultProps} />);
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});

it('matches the snapshot', () => {
  const { container } = render(<Kbd {...defaultProps} />);
  expect(container.firstChild).toMatchSnapshot();
});
