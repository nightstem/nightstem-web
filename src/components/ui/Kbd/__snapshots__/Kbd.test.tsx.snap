// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Kbd > matches the snapshot 1`] = `
<kbd
  class="rounded shadow-2xl border border-foreground/20 p-1.5 leading-[50%]"
>
  Ctrl
</kbd>
`;

exports[`matches the snapshot 1`] = `
<kbd
  class="rounded shadow-2xl border border-foreground/20 p-1.5 leading-[50%]"
>
  Ctrl
</kbd>
`;
